src/lazy/text/buffer.rs

```diff
@@ -998,7 +1019,7 @@ impl<'data> TextBufferView<'data> {
         }
         // Scan ahead to find the end of this struct.
         let struct_body = self.slice_to_end(1);
-        let struct_iter = RawTextStructIterator_1_1::new(struct_body);
+        let struct_iter = RawTextStructIterator_1_0::new(struct_body);
```

Comment:
🗺️ Because `match_ion_struct` was using `RawTextStructIterator_1_1`, it was theoretically possible that Ion 1.0 would accept E-Expressions as long as they were found inside a struct.